test/extensions/tracers/zipkin/span_buffer_test.cc

```diff
@@ -265,102 +303,150 @@ TEST(ZipkinSpanBufferTest, SerializeSpan) {
-            R"("timestamp":"1566058071601051",)"
-            R"("duration":"100",)"
-            R"("localEndpoint":{)"
-            R"("serviceName":"service1",)"
-            R"("ipv4":"AQIDBA==",)"
-            R"("port":8080},)"
-            R"("tags":{)"
-            R"("component":"proxy"})"
-            "}]}",
+  EXPECT_EQ(withDefaultTimestampAndDuration("{"
+                                            R"("spans":[{)"
+                                            R"("traceId":"AAAAAAAAAAE=",)"
+                                            R"("id":"AQAAAAAAAAA=",)"
+                                            R"("kind":"CLIENT",)"
+                                            R"("timestamp":"DEFAULT_TEST_TIMESTAMP",)"
+                                            R"("duration":"DEFAULT_TEST_DURATION",)"
+                                            R"("localEndpoint":{)"
+                                            R"("serviceName":"service1",)"
+                                            R"("ipv4":"AQIDBA==",)"
+                                            R"("port":8080},)"
+                                            R"("tags":{)"
+                                            R"("component":"proxy"}},)"
+                                            R"({)"
+                                            R"("traceId":"AAAAAAAAAAE=",)"
+                                            R"("id":"AQAAAAAAAAA=",)"
+                                            R"("kind":"SERVER",)"
+                                            R"("timestamp":"DEFAULT_TEST_TIMESTAMP",)"
+                                            R"("duration":"DEFAULT_TEST_DURATION",)"
+                                            R"("localEndpoint":{)"
+                                            R"("serviceName":"service1",)"
+                                            R"("ipv4":"AQIDBA==",)"
+                                            R"("port":8080},)"
+                                            R"("tags":{)"
+                                            R"("component":"proxy"})"
+                                            "}]}"),
             serializedMessageToJson<zipkin::proto3::ListOfSpans>(buffer6.serialize()));
 }
 
+TEST(ZipkinSpanBufferTest, TestSerializeTimestampInTheFuture) {
+  ProtobufWkt::Struct objectWithScientificNotation;
+  auto* objectWithScientificNotationFields = objectWithScientificNotation.mutable_fields();
+  (*objectWithScientificNotationFields)["timestamp"] = ValueUtil::numberValue(
+      DEFAULT_TEST_TIMESTAMP); // the value of DEFAULT_TEST_TIMESTAMP is 1584324295476870.
+  const auto objectWithScientificNotationJson =
+      MessageUtil::getJsonStringFromMessage(objectWithScientificNotation, false, true);
+  // Since we use ValueUtil::numberValue to set the timestamp, we expect to
+  // see the value is rendered with scientific notation (1.58432429547687e+15).
+  EXPECT_EQ(R"({"timestamp":1.58432429547687e+15})", objectWithScientificNotationJson);
```

Comment:
Here we have `1.58432429547687e+15` (the input is `1584324295476870`) if we set the field using `ValueUtil::numberValue`.